src/app/[locale]/dataset-list.cy.tsx

```diff
@@ -70,3 +70,174 @@ describe('List rendering', () => {
+
+    cy.getBySel('licensesFilter').within(() => {
+      cy.get('[type="checkbox"]').eq(0).check();
+      cy.get('[type="checkbox"]').eq(1).check();
+    });
+
+    cy.get('@filterRequest').should('have.been.called');
+    cy.getBySel('selectedFilterTag').should('have.length', 2);
+    cy.getBySel('selectedFilterTag')
+      .eq(0)
+      .should('have.text', this.searchResult.filters.licenses[0].name);
+    cy.getBySel('selectedFilterTag')
+      .eq(1)
+      .should('have.text', this.searchResult.filters.licenses[1].name);
+  });
+
+  it('can remove a license filter by deselecting the filter in the sidebar', () => {
+    cy.getBySel('licensesFilter').within(() => {
+      cy.get('[type="checkbox"]').first().check();
+    });
+
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/api/datasets?',
+      },
+      cy.spy().as('filterRequest')
+    );
+
+    cy.getBySel('licensesFilter').within(() => {
+      cy.get('[type="checkbox"]').first().uncheck();
+    });
+
+    cy.get('@filterRequest').should('have.been.called');
+    cy.getBySel('selectedFilterTag').should('have.length', 0);
+  });
+
+  it('can remove a license filter by deselecting it in the selected filter bar', () => {
+    cy.getBySel('licensesFilter').within(() => {
+      cy.get('[type="checkbox"]').first().check();
+    });
+
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/api/datasets?',
+      },
+      cy.spy().as('filterRequest')
+    );
+
+    cy.getBySel('removeSelectedFilter').click();
+
+    cy.get('@filterRequest').should('have.been.called');
+    cy.getBySel('selectedFilterTag').should('have.length', 0);
+  });
+
+  it('can filter by one publisher', function () {
+    cy.intercept(
+      {
+        method: 'GET',
+        url: `/api/datasets?publishers=${encodeURIComponent(
+          this.searchResult.filters.publishers[0].id
+        )}`,
+      },
+      cy.spy().as('filterRequest')
+    );
+
+    cy.getBySel('publishersFilter').within(() => {
+      cy.get('[type="checkbox"]').first().check();
+    });
+
+    cy.get('@filterRequest').should('have.been.called');
+    cy.getBySel('selectedFilterTag').should('have.length', 1);
+    cy.getBySel('selectedFilterTag').should(
+      'have.text',
+      this.searchResult.filters.publishers[0].name
+    );
+  });
+
+  it('can filter based on the search query', () => {
+    cy.intercept(
+      {
+        method: 'GET',
+        url: '/api/datasets?query=My+query',
+      },
+      cy.spy().as('filterRequest')
+    );
+
+    cy.getBySel('searchQuery').type('My query');
+
+    cy.get('@filterRequest').should('have.been.called');
+    cy.getBySel('selectedFilterTag').should('have.length', 1);
+    cy.getBySel('selectedFilterTag').should('have.text', 'My query');
+  });
+
+  it('can filter all categories together (query, license and publisher)', function () {
+    cy.intercept(
+      {
+        method: 'GET',
+        url: `/api/datasets?query=My+query&licenses=${encodeURIComponent(
```

Comment:
The `query` part is manually encoded already, but e.g. `licenses` is not. For consistency: always encode all params in the same way, either manually or by using helpers?